pkg/authz/check/service.go

```diff
@@ -81,22 +80,43 @@ func (svc CheckService) getMatchingSubjects(ctx context.Context, objectType stri
 		return warrantSpecs, err
 	}
 
-	warrants, err = svc.WarrantRepository.GetAllMatchingWildcard(
+	return warrantSpecs, nil
+}
+
+func (svc CheckService) getMatchingSubjectsWithSubjectType(ctx context.Context, objectType string, objectId string, relation string, subjectType string, wntCtx wntContext.ContextSetSpec) ([]warrant.WarrantSpec, error) {
+	log.Debug().Msgf("Getting matching subjects for %s:%s#%s@%s:___%s", objectType, objectId, relation, subjectType, wntCtx)
+
+	warrantSpecs := make([]warrant.WarrantSpec, 0)
+	objectTypeSpec, err := svc.ObjectTypeSvc.GetByTypeId(ctx, objectType)
+	if err != nil {
+		return warrantSpecs, err
+	}
+
+	if _, ok := objectTypeSpec.Relations[relation]; !ok {
+		return warrantSpecs, nil
+	}
+
+	warrants, err := svc.WarrantRepository.GetAllMatchingObjectAndRelationBySubjectType(
 		ctx,
 		objectType,
 		objectId,
 		relation,
+		subjectType,
 		wntCtx.ToHash(),
 	)
 	if err != nil {
-		log.Err(err).Msg("Error fetching warrants matching wildcard")
+		log.Err(err).Msg("Error fetching warrants for object")
```

Comment:
I think we can remove this log line now because all repository methods return an `errors.Wrap`